[DJM] Update DJM docs on the Datadog-Databricks integration (#26242)
DataDog/documentation: content/en/data_jobs/databricks.md

Follow these steps to enable Data Jobs Monitoring for Databricks.

> 1. [Configure the Datadog-Databricks integration](#configure-the-datadog-databricks-integration) for the Databricks workspace.

Reviewer: ```suggestion
1. [Configure the Datadog-Databricks integration](#configure-the-datadog-databricks-integration) for a Databricks workspace.
```

   As an alternative, follow the [official Databricks documentation][10] to generate access token for a [service principal][11].

1. In Datadog, open the Databricks integration tile.
1. On the **Configure** tab, click **Add Databricks Workspace**.
1. Enter a workspace name, your Databricks workspace URL, and the Databricks token you generated.
   {{< img src="data_jobs/databricks/configure-workspace-form.png" alt="In the Datadog-Databricks integration tile, a Databricks workspace is displayed. This workspace has a name, URL, and API token." style="width:100%;" >}}
1. In the **Select products to set up integration** section, make sure the Data Jobs Monitoring product is **Enabled**.
1. In the **Datadog Agent Setup** section, chooose either

>   - [Managed by Datadog](?tab=datadogmanagedglobalinitscriptrecommended#install-the-datadog-agent): Datadog will install and manage the Agent with a global init script in the workspace (recommended).

Reviewer: ```suggestion
  - [Managed by Datadog (recommended)](?tab=datadogmanagedglobalinitscriptrecommended#install-the-datadog-agent): Datadog will install and manage the Agent with a global init script in the workspace.
```

   As an alternative, follow the [official Databricks documentation][10] to generate access token for a [service principal][11].

1. In Datadog, open the Databricks integration tile.
1. On the **Configure** tab, click **Add Databricks Workspace**.
1. Enter a workspace name, your Databricks workspace URL, and the Databricks token you generated.
   {{< img src="data_jobs/databricks/configure-workspace-form.png" alt="In the Datadog-Databricks integration tile, a Databricks workspace is displayed. This workspace has a name, URL, and API token." style="width:100%;" >}}
1. In the **Select products to set up integration** section, make sure the Data Jobs Monitoring product is **Enabled**.

> 1. In the **Datadog Agent Setup** section, chooose to either

Author: I kept the links in the bold portion of the bullets, eg "Managed by Datadog" and "Manually" 

LMK if it should be different.

I removed global in this passage, it seems like a typo since the script is not global in that context

>Follow the instructions below[link] to install and manage the **~global~** init script for installing the agent globally or on specific Databricks clusters.

> 1. In the **Datadog Agent Setup** section, chooose to either

Author: Note the proposed UI changes for the modal 
<img width="2162" alt="config-products" src="https://github.com/user-attachments/assets/f4a1a4b8-539d-44d0-9c91-38207f5e4e94">

  - [Have Datadog install and manage a global init script](?tab=datadogmanagedglobalinitscriptrecommended#install-the-datadog-agent) in the workspace (recommended).
  - Or, [manaully install and manage a global init script](?tab=manuallyinstallaglobalinitscript#install-the-datadog-agent).
  - Or, [manaully install and manage an init script on a specific cluster](?tab=manuallyinstallonaspecificcluster#install-the-datadog-agent).

### Install the Datadog Agent

The Datadog Agent must be installed on Databricks clusters to monitor Databricks jobs that run on the clusters.

> <div class="alert alert-warning">The init script installs the agent when clusters start. All-purpose clusters must be restarted for the init script to install the Datadog Agent.</div>

Reviewer 2: agreed ^ also a note to capitalize Agent when referring to the Datadog Agent

> <div class="alert alert-warning">The init script installs the agent when clusters start. All-purpose clusters must be restarted for the init script to install the Datadog Agent.</div>

Author: Done

1. Click **Select API Key** to either select an existing Datadog API key or create a new Datadog API key.
   {{< img src="data_jobs/databricks/configure-data-jobs-monitoring-new.png" alt="In the Datadog-Databricks integration tile, Datadog Agent Setup when adding a Databricks workspace. Datadog can install and manage a global init script." style="width:100%;" >}}

#### To modify a Databricks workspace account in the Datadog-Databricks integration

1. On the **Configure** tab, click the workspace in the list of workspaces
1. Click the **Configured Products** tab
1. Make sure the Data Jobs Monitoring product is **Enabled**.
1. In the **Datadog Agent Setup** section, select the **Managed by Datadog** toggle button.
1. Click **Select API Key** to either select an existing Datadog API key or create a new Datadog API key.
   {{< img src="data_jobs/databricks/configure-data-jobs-monitoring-existing.png" alt="In the Datadog-Databricks integration tile, Datadog Agent Setup for a Databricks workspace already added to the integration. Datadog can install and manage a global init script." style="width:100%;" >}}

{{% /tab %}}

{{% tab "Manually install a global init script" %}}

> When adding a new Databricks workspace:

Reviewer: @tiffon remove all of this new next section talking about the integration tile first as part of the manual instructions. I see the logic of why you put it here for completeness assuming the user is starting at the tile, but everything they need to actually do to complete this is in the Databricks console, so we should just tell them to go there from the start and simplify this. I would be annoyed as a user following these steps, and going to the tile, only to not have anything to do/change there.

[2]: /getting_started/site/

{{% /tab %}}

{{% tab "Manually install on a specific cluster" %}}

> When adding a new Databricks workspace:

Reviewer: @tiffon remove all of this new next section talking about the integration tile first as part of the manual instructions. I see the logic of why you put it here for completeness assuming the user is starting at the tile, but everything they need to actually do to complete this is in the Databricks console, so we should just tell them to go there from the start and simplify this. I would be annoyed as a user following these steps, and going to the tile, only to not have anything to do/change there.

{{< /tabs >}}

> ### Restart existing clusters

Reviewer: ```suggestion
### Restart currently running clusters
```

The init script installs the Agent when clusters start.

> Existing clusters, like all-purpose clusters or long-lived job clusters, must be restarted for the init script to install the Datadog Agent.

Reviewer: ```suggestion
Currently running all-purpose clusters or long-lived job clusters must be manually restarted for the init script to install the Datadog Agent.

For scheduled jobs that run on job clusters, the init script gets installed automatically on the next run.
```

> 

Reviewer: ```suggestion
```

> All new clusters that start will use the init script (if the script is global or configured for the job cluster).

Reviewer: ```suggestion
```

Author: >```
>For scheduled jobs that run on job clusters, the init script gets installed automatically on the next run.
>```

@warrierr Why make this comment about new clusters starting specific to jobs? Any new node that comes online (IIUC) will use the now installed init script to install the agent, even nodes added due to auto-sizing up on existing clusters (I would guess anyway).

Reviewer: I added this because I think it clears up to customers what they need to take action on. In the global init script case, it doesn't make sense to go update the job cluster config for batch Databricks jobs since it will get the global init on the next run. So customers only need to take action to restart streaming job clusters and all-purpose clusters.

> Any new node that comes online (IIUC) will use the now installed init script to install the agent, even nodes added due to auto-sizing up on existing clusters
True, but to confidently get all nodes to have the agent, and know it is in a good state, you need to restart the cluster. Not sure adding the technical nuance here around how it can partially get added to new nodes because of autoscaling is necessary.

Author: Change applied with 

>For scheduled jobs that run on job clusters, the init script **installs the Datadog Agent** automatically on the next run.

### Aggregate cluster metrics from one-time job runs
   This configuration is applicable if you want cluster resource utilization data about your jobs and create a new job and cluster for each run via the [one-time run API endpoint][8] (common when using orchestration tools outside of Databricks such as Airflow or Azure Data Factory).

>    If you are submitting Databricks Jobs via the [one-time run API endpoint][8], each job run has a unique job ID. This can make it difficult to group and analyze cluster metrics for jobs that use ephemeral clusters. To aggregate cluster utilization from the same job and assess performance across multiple runs, you must set the `DD_JOB_NAME` variable inside the `spark_env_vars` of every `new_cluster` to the same value as your request payload's `run_name`.

Reviewer 2: ```suggestion
   If you are submitting Databricks Jobs through the [one-time run API endpoint][8], each job run has a unique job ID. This can make it difficult to group and analyze cluster metrics for jobs that use ephemeral clusters. To aggregate cluster utilization from the same job and assess performance across multiple runs, you must set the `DD_JOB_NAME` variable inside the `spark_env_vars` of every `new_cluster` to the same value as your request payload's `run_name`.
```
